src/repositoryModals.jsx

```diff
@@ -0,0 +1,451 @@
+
+import * as remotes from './remotes';
+
+const _ = cockpit.gettext;
+
+export const RemoveRepositoryModal = ({ origin, availableRemotes, refreshRemotes }) => {
+    const Dialogs = useDialogs();
+    const [error, setError] = useState('');
+    const [selectedRemotes, setSelectedRemotes] = useState([]);
+
+    const onDelete = () => {
+        Promise.all(selectedRemotes.map(remote => remotes.deleteRemote(remote)))
+                .then(() => refreshRemotes())
+                .then(Dialogs.close, ex => setError(ex.message));
+    };
+
+    const handleChange = (event, isChecked) => {
+        if (isChecked) {
+            setSelectedRemotes([...selectedRemotes, event.target.id]);
+        } else {
+            setSelectedRemotes(selectedRemotes.filter(remote => remote !== event.target.id));
+        }
+    };
+
+    const actions = [
+        <Button key="remove-repo"
+            variant="danger"
+            isDisabled={selectedRemotes.length === 0}
+            onClick={() => onDelete()}>
+            {_("Remove")}
+        </Button>,
+        <Button key="cancel" variant="link" onClick={Dialogs.close}>
+            {_("Cancel")}
+        </Button>
+    ];
+
+    const repositories = availableRemotes.map(remote => {
+        return (
+            <Checkbox label={remote}
+                key={remote}
+                id={remote}
+                onChange={handleChange}
+                isChecked={selectedRemotes.includes(remote)}
+                isDisabled={origin.remote === remote} />
+        );
+    });
+
+    return (
+        <Modal isOpen
+            id="remove-repository-modal"
+            title={_("Remove repository")}
+            position="top"
+            variant="small"
+            onClose={Dialogs.close}
+            actions={actions}
+        >
+            <Form isHorizontal>
+                {error && <Alert variant="danger" isInline title={error} />}
+                <FormGroup isStack
+                    label={_("Repository")}
+                    role="group"
+                    hasNoPaddingTop
+                >
+                    {repositories}
+                </FormGroup>
+            </Form>
+        </Modal>
+    );
+};
+
+RemoveRepositoryModal.propTypes = {
+    origin: PropTypes.object.isRequired,
+    availableRemotes: PropTypes.array.isRequired,
+    refreshRemotes: PropTypes.func.isRequired,
+};
+
+export const AddRepositoryModal = ({ refreshRemotes }) => {
+    const Dialogs = useDialogs();
+    const [newRepoName, setNewRepoName] = useState("");
+    const [newRepoURL, setNewRepoURL] = useState("");
+    const [newRepoTrusted, setNewRepoTrusted] = useState(false);
+
+    const [hasValidation, setHasValidation] = useState(false);
+    const [addNewRepoError, setAddNewRepoError] = useState(undefined);
+
+    const onAddRemote = () => {
+        if (!(newRepoURL.trim().length && newRepoName.trim().length)) {
+            setHasValidation(true);
+            return;
```

Comment:
These 3 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-403-20231124-162305-47d51597-fedora-coreos-devel//Coverage/lcov/github-pr.diff.gcov.html)